examples/express-ts/typings/objection/index.d.ts

```diff
@@ -0,0 +1,692 @@
+    $id(id: any): void;
+
+    $beforeValidate(jsonSchema: JsonSchema, json: Object, opt: ModelOptions): JsonSchema;
+    $validate(): void // may throw ValidationError if validation fails
+    $afterValidate(json: Object, opt: ModelOptions): void; // may throw ValidationError if validation fails
+
+    $toDatabaseJson(): Object;
+    $toJson(): Object;
+    toJSON(): Object;
+    $parseDatabaseJson(json: Object): Object;
+    $formatDatabaseJson(json: Object): Object;
+    $parseJson(json: Object, opt?: ModelOptions): Object;
+    $formatJson(json: Object): Object;
+    $setJson(json: Object, opt?: ModelOptions): this;
+    $setDatabaseJson(json: Object): this;
+
+    $set(obj: Object): this;
+    $omit(keys: string | string[] | Properties): this
+    $pick(keys: string | string[] | Properties): this;
+    $clone(): this;
+
+    $query(): QueryBuilder;
+    $relatedQuery(relationName: string, transaction?: Transaction): QueryBuilder;
+
+    $loadRelated(expression: RelationExpression, filters?: Filters): QueryBuilder;
+
+    $traverse(traverser: TraverserFunction): void;
+    $traverse<T extends Model>(this: ModelClass<T>, filterConstructor: ModelClass<T>, traverser: TraverserFunction): void;
+
+    $knex(): knex;
+    $transaction(): knex; // TODO: this is based on the documentation, but doesn't make sense (why not Transaction?)
+
+    $beforeInsert(queryContext: Object): Promise<any> | void;
+    $afterInsert(queryContext: Object): Promise<any> | void;
+    $afterUpdate(opt: ModelOptions, queryContext: Object): Promise<any> | void;
+    $beforeUpdate(opt: ModelOptions, queryContext: Object): Promise<any> | void;
+  }
+
+  export class QueryBuilder {
+    static extend(subclassConstructor: FunctionConstructor): void;
+    static forClass(modelClass: Model): QueryBuilder;
+  }
+
+  export interface QueryBuilder extends QueryInterface, Promise<any> {
+
+    findById(idOrIds: IdOrIds): QueryBuilder;
+
+    insert(modelsOrObjects: ModelsOrObjects): QueryBuilder;
+    insertAndFetch(modelsOrObjects: ModelsOrObjects): QueryBuilder;
+
+    insertGraph(modelsOrObjects: ModelsOrObjects): QueryBuilder;
+    insertGraphAndFetch(modelsOrObjects: ModelsOrObjects): QueryBuilder;
+
+    insertWithRelated(graph: ModelsOrObjects): QueryBuilder;
+    insertWithRelatedAndFetch(graph: ModelsOrObjects): QueryBuilder;
+
+    update(modelOrObject: Object | Model): QueryBuilder;
+    updateAndFetchById(id: Id, modelOrObject: Object | Model): QueryBuilder;
+
+    patch(modelOrObject: Object | Model): QueryBuilder;
+    patchAndFetchById(id: Id, modelOrObject: Object | Model): QueryBuilder;
+    patchAndFetch(modelOrObject: Object | Model): QueryBuilder;
+
+    deleteById(idOrIds: IdOrIds): QueryBuilder;
+
+    relate(ids: IdOrIds | ModelsOrObjects): QueryBuilder;
+    unrelate(): QueryBuilder;
+
+    forUpdate(): QueryBuilder;
+    forShare(): QueryBuilder;
+
+    // TODO: fromJS does not exist in current knex documentation: http://knexjs.org/#Builder-fromJS
```

Comment:
I think from JS was removed from knex at some point... I had never used it either.